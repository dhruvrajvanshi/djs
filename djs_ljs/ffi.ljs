
/**
 * Null terminated C style bytestring
 * @example
 * const msg: *char = str`Hello` // ['H', 'e', 'l', 'l', '0', '\0']
 */
export const c_str = @builtin("c_str")

/**
 * Equivalent to the c char type.
 * Can be either i8 or u8, depending on the platform
 */
export type c_char = @builtin("c_char")


export type c_int = @builtin("c_int")
