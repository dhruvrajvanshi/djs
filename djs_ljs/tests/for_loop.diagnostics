ERROR: ./tests/for_loop.ljs:6: TODO: PostIncrement cannot be inferred at the moment
6|      for (let i = 0; i < 10; i++) {
                                ^~~
7|          puts(ffi.c_str`LOOP`)
8|      }