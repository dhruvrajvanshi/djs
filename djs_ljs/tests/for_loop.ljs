import * as ffi from "../ffi.ljs"

extern function puts(s: *ffi.c_char): ffi.c_int

function main(): ffi.c_int {
    for (let i = 0; i < 10; i++) {
        puts(ffi.c_str`LOOP`)
    }
    puts(ffi.c_str`PASS`)
    return 0
}
