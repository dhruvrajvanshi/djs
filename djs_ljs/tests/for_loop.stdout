LOOP
LOOP
LOOP
LOOP
LOOP
LOOP
LOOP
LOOP
LOOP
LOOP
PASS
